fastapi_sqla/aws_rds_iam_support.py

```diff
@@ -0,0 +1,25 @@
+import boto3
+from pydantic import BaseSettings
+
+
+def get_authentication_token(host, port, user):
+    session = boto3.Session()
+    client = session.client("rds")
```

Comment:
What's involved in creating the client? Is it cheap? If it's not it _might_ be worth caching it.

Reply:
Good question. Since it might be called often, caching is probable worth it.

Reply:
```
❯ python -m timeit -n 1000 'import boto3;session = boto3.Session();client = session.client("rds")'
1000 loops, best of 5: 55.8 msec per loop
❯ python -m timeit -n 1000 'dict(a=1, b=2, c=3, d=4)'                                                                                                                                                  
1000 loops, best of 5: 161 nsec per loop
```

Takes at best 55ms.

~[ ](https://media.giphy.com/media/l0MYNI50nCCOa3f8c/giphy.gif)

Reply:
But IIUC, it is only called when connecting.

Reply:
Which occurs at startup.

Reply:
I believe it is called whenever a new connection to the DB is established which would occur during the lifetime of the service as well as more DB connections are needed or as existing ones are disconnected for any number of reasons.

Reply:
55ms to establish a new connection is not great, but not a big deal. Since caching correctly is hard it's probably best to leave it as is for now and readjust if we se a problem.

Reply:
> I believe it is called whenever a new connection to the DB is established

True. 
By default sqla connection pool size is 5 with a overflow max of 10. Also, by default, all connections are recycled.

https://docs.sqlalchemy.org/en/14/core/engines.html#sqlalchemy.create_engine

Reply:
We should probably bump the default a bit as it bites us in the past but problem for later I suppose.